include/fw_info.h

```diff
@@ -78,75 +60,156 @@ struct __packed fw_info {
 				#member " has wrong offset")
 
 /* Static asserts to ensure compatibility */
-OFFSET_CHECK(struct fw_info, magic, 0);
-OFFSET_CHECK(struct fw_info, firmware_size, CONFIG_FW_INFO_MAGIC_LEN);
-OFFSET_CHECK(struct fw_info, firmware_version,
-	(CONFIG_FW_INFO_MAGIC_LEN + 4));
-OFFSET_CHECK(struct fw_info, firmware_address,
-	(CONFIG_FW_INFO_MAGIC_LEN + 8));
+OFFSET_CHECK(struct fw_info_abi, abi_len, 0);
+OFFSET_CHECK(struct fw_info_abi, abi_id, 4);
+OFFSET_CHECK(struct fw_info_abi, abi_flags, 8);
+OFFSET_CHECK(struct fw_info_abi, abi_version, 12);
+/** @endcond
+ */
+
+
+/**
+ * This struct is used to request an ABI. It contains a pointer to a non-
+ * initialized pointer that a previous boot stage will populate with a pointer
+ * to a compliant ABI in another image. An ABI fulfills a request if the ID
```

Comment:
Not easy to follow the difference and connection between the "previous boot stage", "another image" and "the image" in this explaination

Reply:
Rewrote this section